ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -26,65 +29,140 @@ Accepted
 ```yaml
-apiVersion: v1
-kind: Secret
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
 metadata:
-  name: tmp-jdbc-connection-parameter-upload
-  labels:
-    spi.appstudio.redhat.com/upload-secret: secret
-    spi.appstudio.redhat.com/target-environment: prod
-    spi.appstudio.redhat.com/target-secret-name: jdbc-connection-parameter
-type: Opaque
-data:
- ...
+    namespace: jdoe-tenant
+spec:
+    target:
+        - namespace: spi-test-target
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
+
 ```
+
+#### Example: If RemoteSecret has to be created and uploaded without setting any target
+
 ```yaml
 apiVersion: appstudio.redhat.com/v1beta1
 kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter-prod
+    namespace: jdoe-tenant
 spec:
-  secret:
-    environment: prod
-    name: jdbc-connection-parameter
- ...
+    target:
+status:
+  TBD
 ```
 
-#### Example: If the destination is `Namespace` and secret has to be linked to SA
+#### Example: If RemoteSecret has to be created with multiple target namespaces
 ```yaml
-apiVersion: v1
-kind: Secret
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter
-  annotations:
-      spi.appstudio.redhat.com/secret-linkedto-serviceAccount-name: sa-star
-  labels:
-    spi.appstudio.redhat.com/upload-secret: secret
-    spi.appstudio.redhat.com/target-namespace: ns-9459039345
-type: Opaque
-data:
- ...
+    name: test-remote-secret-secret
+spec:
+    target:
+        - namespace: jdoe-test-ns
+          secretName: test-remote-secret-secret-5bmq9
+        - namespace: jdoe-prod-ns
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
+```
+
+#### Example: If RemoteSecret has to be created with multiple target namespaces including BYC use-case
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+spec:
+    target:
+        - namespace: jdoe-test-ns
+          secretName: test-remote-secret-secret-5bmq9
+        - namespace: jdoe-prod-ns
+          secretName: test-remote-secret-secret-5bmq9
+          apiURL: https://somedomain.copm:443
+          clusterCredentialsSecret: team-a--prod-dtc--secret
+status:
+    TBD
 ```
+
+#### Example: If RemoteSecret has to be created with target namespace and Environment
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+    labels:
+        appstudio.redhat.com/remotesecret-target-environment: prod
+        appstudio.redhat.com/remotesecret-target-component: m-service
+        appstudio.redhat.com/remotesecret-target-application: coffee-shop
+spec:
+    target:
+        - namespace: jdoe-tenant
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
+```
+
+#### Example: If RemoteSecret has to be created all Environments of certain component and application
```

Comment:
If we are keeping Image Controller in context, then, probably, the credentials to access Component image repository should be propagated into all environments (even ones created after image repository provision). Therefore, this use-case should be primary for Image Controller.

Reply:
Thank you